ubuntuHomestead.py

```diff
@@ -50,86 +51,122 @@
-URL_NAME = "homestead.app"
-DEFAULT_FRAMEWORK_URL = "https://github.com/laravel/laravel.git"
-DEFAULT_FW_DIR_NAME = 'Laravel'
-NUMBER_OF_CPUS = '1'
 
 # System modifications
 sys.stdin = open('/dev/tty')
 
 
 def md5(fname):
-	hash_md5 = hashlib.md5()
-	with open(fname, "rb") as f:
-		for chunk in iter(lambda: f.read(4096), b""):
-			hash_md5.update(chunk)
-	return hash_md5.hexdigest()
+    hash_md5 = hashlib.md5()
+    with open(fname, "rb") as f:
+        for chunk in iter(lambda: f.read(4096), b""):
+            hash_md5.update(chunk)
+    return hash_md5.hexdigest()
 
 
 def install(program_name, common_name):
-	print('Checking whether {} is installed'.format(common_name))
-	if shutil.which(program_name) is None:
-		print('{} is not installed, installing now'.format(common_name))
-		os.system('sudo apt install {} -y'.format(program_name))
-	else:
-		print('{} is already installed'.format(common_name))
-
-	time.sleep(1)
-	os.system('clear')
-
-
-help_out = 'This provides a small list of command line arguments that this \n' \
-		   'program will accept:\n' \
-		   '\t-u\tEnter a custom url to fetch a Laravel repository from\n' \
-		   '\t-n\tEnter a custom application name\n' \
-		   '\t-d\tEnter a custom directory path to install Laravel in\n' \
-		   '\t-D\tEnter a custom name for your Laravel directory\n' \
-		   '\t-c\tEnter the number of cores to assign to your VM\n' \
-		   '\t-h\tHelp screen'
-
-try:
-	opts, args = getopt.getopt(sys.argv[1:], "hu:n:l:d:D:c:", ['help'])
-except getopt.GetoptError:
-	print('names.py -f -l -e -d -n <number of names to print>')
-	sys.exit(2)
-
-for opt, arg in opts:
-	if opt in ('-h', '--help'):
-		print(help_out)
-		sys.exit(1)
-	elif opt in '-u':
-		DEFAULT_FRAMEWORK_URL = arg
-		if 'git@github.com' in DEFAULT_FRAMEWORK_URL:
-			SSH_LINK = True
-	elif opt in '-n':
-		URL_NAME = arg + '.app'
-	elif opt in '-D':
-		DEFAULT_FW_DIR_NAME = arg
-	elif opt in '-d':
-		FRAMEWORK_PATH = arg
-	elif opt in '-c':
-		NUMBER_OF_CPUS = arg
-
-for x in ENV_VARS:
-	if x in os.environ:
-		if x == 'SCRIPTS_URL':
-			DEFAULT_FRAMEWORK_URL = os.getenv(x)
-			if 'git@github.com' in DEFAULT_FRAMEWORK_URL:
-				SSH_LINK = True
-		elif x == 'SCRIPTS_URL_NAME':
-			URL_NAME = os.getenv(x) + '.app'
-		elif x == 'SCRIPTS_DIR_NAME':
-			DEFAULT_FW_DIR_NAME = os.getenv(x)
-		elif x == 'SCRIPTS_DIR_PATH':
-			FRAMEWORK_PATH = os.getenv(x)
-		elif x == 'SCRIPTS_CORES':
-			NUMBER_OF_CPUS = os.getenv(x)
+    print('Checking whether {} is installed'.format(common_name))
+    if shutil.which(program_name) is None:
+        print('{} is not installed, installing now'.format(common_name))
+        os.system('sudo apt install {} -y'.format(program_name))
+    else:
+        print('{} is already installed'.format(common_name))
+
+    time.sleep(1)
+    os.system('clear')
+
+def add_ssh_key_to_github(ssh_key):
+    # Check if your key already works.
+    ssh_accepted = os.system('ssh -Ta git@github.com')
+
+    while not ssh_accepted:
```

Comment:
So, I know the variable was in there when you started, but now that we are actually assigning a value to it, it should be checked thoroughly. Before, it was setup with a Boolean value that would cause the program to run regardless. Now it is checking for a Boolean, but the return from the os.syst... command is an integer, whether it succeeds or fails.
![Example output](https://user-images.githubusercontent.com/17666816/30789037-6c1aa182-a157-11e7-93d3-9521e65a0c32.png)